### ISSUE TITLE
Histogram ignores `xmax` if `xmax` is greater than the largest `x` value being plotted

### ISSUE THREAD

**Issue by alexpeters1208** [deephaven/web-client-ui#1667]
**Description**

Histograms created with `plot_xy_hist` do not extend their x-axis to `xmax` if this value is larger than the maximum x value present in the dataset.

**Steps to reproduce**

Run this code:
```
from deephaven.plot.figure import Figure
from deephaven import new_table
from deephaven.column import int_col

source = new_table([
    int_col("Values", [1, 2, 2, 3, 3, 3, 4, 4, 5])
])

new_plot = Figure()\
    .plot_xy_hist(series_name="Histogram Values", t=source, x="Values", nbins=5, xmin=0.0, xmax=10.0)\
    .chart_title(title="Histogram of Values")\
    .show()
```
**Expected results**

I expect this histogram to have data between 1 and 5, and an x-axis that extends out to 10.

**Actual results**

The axis stops at 6. It seems to want to obey `nbins`, and is binning the data _as though_ the plot extends out to 10, but only non-zero bins are being shown, so there are fewer `nbins` than requested.

<img width="1113" alt="Screenshot 2023-12-05 at 10 29 59 AM" src="https://github.com/deephaven/deephaven-core/assets/80283343/4e3a0877-52ba-44e4-a7d8-a08be85abfcc">

**Versions**
 - Deephaven: 0.30.1
 - OS: MacOS
 - Browser: Safari

**Comment by mofojed** [comment]
I think the API may be confusing - this is a maximum x value, not the maximum x range to show in the rendered plot.
Even the example in the docs shows an example with `nbins=5` and only 3 bins show up: https://deephaven.io/core/docs/how-to-guides/plotting/histogram/#from-a-table
![image](https://github.com/deephaven/web-client-ui/assets/4505624/156fe593-56b4-4fda-9f15-83c9b093a5b3)

**Comment by alexpeters1208** [comment]
> I think the API may be confusing - this is a maximum x value, not the maximum x range to show in the rendered plot.
> Even the example in the docs shows an example with `nbins=5` and only 3 bins show up: https://deephaven.io/core/docs/how-to-guides/plotting/histogram/#from-a-table
> ![image](https://private-user-images.githubusercontent.com/4505624/289932292-156fe593-56b4-4fda-9f15-83c9b093a5b3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI2NjYzMjMsIm5iZiI6MTcwMjY2NjAyMywicGF0aCI6Ii80NTA1NjI0LzI4OTkzMjI5Mi0xNTZmZTU5My01NmI0LTRmZGEtOWYxNS04M2M5YjA5M2E1YjMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQUlXTkpZQVg0Q1NWRUg1M0ElMkYyMDIzMTIxNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyMzEyMTVUMTg0NzAzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YjFiMjg0YjYwOTFjYWViOTc0ZWNkMWMyMGNiMWY0Nzc3MGU4NDI1NWRjNzYyZDg4OGNiYzAwM2VlMGU4OWNhNiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.MNgQJGak2lpgIDJPR8bcB4YPT0utBiySnbiM7tOw9GI)

Yep, I'd say that is a confusing API and not what I expected / wanted. Not sure what the resolution here should be.
